src/renderer/actions/connections.ts

```diff
@@ -126,13 +135,13 @@ export function connect(id, databaseName, reconnecting = false, sshPassphrase) {
       }
       const currentConn = getState().connections;
       if (!currentConn.databases.length) {
-        this.disconnect();
+        dispatch(disconnect());
```

Comment:
Found 2 issues here:
1. It was calling this.disconnect, which does not exist, instead of disconnect
2. It was not dispatching the event from the disconnect execution